app/hotels/src/singleHotel/roomPicker/RoomPicker.js

```diff
@@ -71,34 +29,110 @@ export default function RoomPicker({
 }: Props) {
   if (selectedCount === 0) {
     return (
-      <NoneSelected price={price} currency={currency} onSelect={increment} />
+      <TouchableItem
+        style={[
+          styles.container,
+          styles.borderRadiusLeft,
+          styles.borderRadiusRight,
+        ]}
+        onPress={increment}
+      >
+        <View style={styles.row}>
+          <ButtonText style={styles.text} text="Select" />
+          <View style={styles.row} />
+          <Price amount={price} currency={currency} style={styles.text} />
+        </View>
+      </TouchableItem>
     );
   }
 
   return (
-    <View style={styles.container}>
-      <TouchableOpacity
-        style={[styles.button, styles.minusButton]}
-        onPress={decrement}
-      >
-        <Text style={styles.buttonText}>-</Text>
-      </TouchableOpacity>
-      <View style={styles.priceAndCount}>
-        <Text style={styles.count}>{selectedCount} × </Text>
-        <Price amount={price} currency={currency} style={styles.price} />
+    <View style={styles.row}>
+      <MinusButton onPress={decrement} />
+
+      <View style={[styles.container, styles.row, styles.priceAndCount]}>
+        <Text style={[styles.text, styles.count]}>{selectedCount} × </Text>
+        <Price amount={price} currency={currency} style={styles.text} />
       </View>
-      {selectedCount < selectableCount ? (
-        <TouchableOpacity
-          style={[styles.button, styles.plusButton]}
-          onPress={increment}
-        >
-          <Text style={styles.buttonText}>+</Text>
-        </TouchableOpacity>
-      ) : (
-        <View style={[styles.button, styles.plusButton, styles.disabledButton]}>
-          <Text style={styles.buttonText}>+</Text>
-        </View>
-      )}
+
+      <PlusButton
+        onPress={increment}
+        disabled={selectedCount >= selectableCount}
+      />
     </View>
   );
 }
+
+const MinusButton = ({ onPress }) => {
+  return (
+    <TouchableItem
+      style={[styles.button, styles.borderRadiusLeft]}
+      onPress={onPress}
+    >
+      <Text style={styles.buttonText}>-</Text>
+    </TouchableItem>
+  );
+};
+
+const PlusButton = ({ disabled, onPress }) => {
+  const child = <Text style={styles.buttonText}>+</Text>;
+  const buttonStyles = [styles.button, styles.borderRadiusRight];
+
+  if (disabled) {
+    return (
+      <View style={[...buttonStyles, styles.disabledButton]}>{child}</View>
+    );
+  }
+
+  return (
+    <TouchableItem style={buttonStyles} onPress={onPress}>
```

Comment:
There is a disabled prop `<TouchableItem style={buttonStyles} onPress={onPress} disabled={disabled}>`

Reply:
Good point - even better. I just copy pasted the previous 💩 solution... :)

Reply:
@tbergq Should be fixed.